### DISCUSSION TITLE
iOS / Android Apps

### DISCUSSION THREAD

**Discussion by countrymusicfy** [AzuraCast/AzuraCast#4158] [Q&A]
Does anyone have any suggestions on building mobile apps for an azuracast station that is cheap and easy?

**Comment by ghost** [comment]
#4132

**Comment by ivellios1988** [comment]
Last week I built a PWA player for my radio: https://player-pwa.paranormalium.pl/

It uses a template I bought on Envato market, the only thing I had to do to make it work as installable PWA was to create a set of icons, create a manifest.json file and put some HTML code into index.php. A disadvantage of PWA apps is that they cannot play streams automatically after being launched, instead, they require user action, for example, click on Play button. This is a limitation introduced in all major browsers and it seems to be impossible to override :( On the other side, once you have a PWA app, it will work practically on every smartphone and you won't need to bother adding it to Google Play, Apple Store, etc. app stores.

I also have an Android app for my radio, it's based on a built template as well. Here the task was more difficult because I had to learn to use Android Studio a little bit.
